### PR TITLE
feat(endpoint): register endpoint with Endpoint.of injection

### PR DESCRIPTION
**Pull request by doktordirk**
This change is [Reviewable](https://reviewable.io/reviews/spoonx/aurelia-api/104)

### CODE REVIEW

**Comment by doktordirk** [comment]
@RWOverdijk sortof to nice to leave it out of rc3 imho

File: src/endpoint.js [hunk 1]
```diff
@@ -5,33 +5,42 @@ import {Config} from './config';
...
    */
   get(container) {
-    return container.get(Config).getEndpoint(this._key);
+    let config = container.get(Config);
+    let endpoint = config.getEndpoint(this._key);
+
+    if (endpoint) return endpoint;
+
+    config.registerEndpoint(this._key, ...this._rest);
+    return config.getEndpoint(this._key);
```
Comment (RWOverdijk) [inline src/endpoint.js]: Newline before return

File: src/endpoint.js [hunk 2]
```diff
@@ -5,33 +5,42 @@ import {Config} from './config';
...
+   * Resolve for key. If no endpoint is found, create new one with config.registerEndpoint(key, ...rest)
    *
    * @param {Container} container
    *
    * @return {*}
    */
   get(container) {
-    return container.get(Config).getEndpoint(this._key);
+    let config = container.get(Config);
+    let endpoint = config.getEndpoint(this._key);
```
Comment (RWOverdijk) [inline src/endpoint.js]: Align consecutive assignments

File: src/endpoint.js [hunk 3]
```diff
@@ -5,33 +5,42 @@ import {Config} from './config';
...
+   * Get a new resolver for `key` and optional '...rest' parameters for optional dynamic endpoint registration with config.registerEndpoint(key, ...rest)
    *
    * @param {string} key
+   * @param {[args]} ...rest
    *
    * @return {Endpoint}
    */
-  static of(key) {
-    return new Endpoint(key);
+  static of(key, ...rest) {
```
Comment (RWOverdijk) [inline src/endpoint.js]: What's rest needed for?

File: src/endpoint.js [hunk 4]
```diff
@@ -5,33 +5,42 @@ import {Config} from './config';
...
    * @param {Container} container
    *
    * @return {*}
    */
   get(container) {
-    return container.get(Config).getEndpoint(this._key);
+    let config = container.get(Config);
+    let endpoint = config.getEndpoint(this._key);
+
+    if (endpoint) return endpoint;
```
Comment (RWOverdijk) [inline src/endpoint.js]: Curly braces

**Comment by doktordirk** [comment]
hmm, it might get to messy and it's very programmers-error prone. in case, someone needs to set endpoints in an encapsulated module, a separated Config instance would be much saver. 
closed

**Pull request closed without merging**